### PR TITLE
Add reasons to 2.6.x

### PR DESCRIPTION
**Pull request by TartanLlama**

### CODE REVIEW

File: sg20/d1389/d1389.bs [hunk 1]
```diff
@@ -323,27 +323,46 @@ TODO (why?)
...
-### [tools.debugger] Introduce a debugger early
+### [tools.testing] Introduce a testing framework
+
+Examples: [Catch2](https://github.com/catchorg/Catch2), [Google Test](https://github.com/google/googletest)
 
 TODO (why?)
 
+### [tools.debugger] Introduce a debugger early
+
+The ability to step through running code and examine execution state will enable students to troubleshoot issues and correct assumptions they have made about the behavior of language and library features.
```
Comment (cjdb) [inline sg20/d1389/d1389.bs]: Should there be recommendations for debuggers? I'm only aware of four:
* GDB
* LLDB
* Visual Studio's debugger
* windbg (not for beginners?)

Are there others that should be considered or avoided?

File: sg20/d1389/d1389.bs [hunk 2]
```diff
@@ -310,7 +310,7 @@ TODO (what?, why?, how?, where?, when?, how many?)
 * Most recent Clang: version 7
 * Most recent MSVC: version 2017 (with updates)
 
-TODO (why?)
+Modern compilers have significantly improved error messages, making it easier for novices to find and correct their errors.
```
Comment (cjdb) [inline sg20/d1389/d1389.bs:313]: Please terminate lines at 100 characters.

File: sg20/d1389/d1389.bs [hunk 3]
```diff
@@ -323,27 +323,46 @@ TODO (why?)
...
+### [tools.debugger] Introduce a debugger early
+
+The ability to step through running code and examine execution state will enable students to troubleshoot issues and correct assumptions they have made about the behavior of language and library features.
+
 ### [tools.package.management] Use a package manager
 
-TODO (why?)
+Examples: [Vcpkg](https://github.com/Microsoft/vcpkg), [Conan](https://conan.io/)
+
+Downloading, installing, and building against dependencies in C++ can be a challenge, especially for beginners. Package managers help alleviate this by providing tested packages along with automatic installation scripts.
```
Comment (cjdb) [inline sg20/d1389/d1389.bs]: +1

You don't need to do this, but I think a section discussing the pros and cons of both (or linking to a resource that does so) would be beneficial.

File: sg20/d1389/d1389.bs [hunk 4]
```diff
@@ -323,27 +323,46 @@ TODO (why?)
...
+Online compilers are invaluable tools for communicating about small snippets of code. Depending on the tool, they let the user compile programs using multiple toolchains, check the output of their code, and share the snippets with others.
+
+Compiler Explorer's live updates can be particularly useful when experimenting with new features. The assembly view could overwhelm students however, so care should be taken when introducing this tool.
 
 ### [tools.code.formatter] Use a code formatter
 
 Examples: clang-format
 
-TODO (why?)
+Choosing a code formatter and picking a canonical style (it doesn't really matter which one) will avoid some code style arguments and improve uniformity among student's code. The latter will make marking and comparing solutions easier.
```
Comment (cjdb) [inline sg20/d1389/d1389.bs]: +++++

**Review by cjdb — CHANGES_REQUESTED** [review]

**Comment by TartanLlama** [comment]
Made changes.

**Pull request merged**